### ISSUE TITLE
diffing update should be default

### ISSUE THREAD

**Issue by kneth** [realm/realm-js#2448]
Currently the default behaviour of [`Realm.create()`](https://realm.io/docs/javascript/latest/api/Realm.html#create) is to create new objects (`updateMode` defaults to `'never'`). The suggestion is to change it to `'modified'`.

**Comment by astigsen** [comment]
I agree that we should change the default to do the diffing. It is a breaking change, but this is a good opportunity to introduce it.

**Comment by cmelchior** [comment]
We should coordinate that between all bindings IMO, so we all agree on what the default behavior is. That said I agree that diffed updates are probably the behavior that most people expect.
